java/repositories.bzl

```diff
@@ -25,472 +25,339 @@
 load("@bazel_tools//tools/build_defs/repo:utils.bzl", "maybe")
+load("//toolchains:local_java_repository.bzl", "local_java_repository")
 load("//toolchains:remote_java_repository.bzl", "remote_java_repository")
 
 def java_tools_javac11_repos():
-    maybe(
-        http_archive,
+    http_archive(
+        name = "remote_java_tools",
+        sha256 = "b763ee80e5754e593fd6d5be6d7343f905bc8b73d661d36d842b024ca11b6793",
+        urls = [
+            "https://mirror.bazel.build/bazel_java_tools/releases/java/v11.5/java_tools-v11.5.zip",
+            "https://github.com/bazelbuild/java_tools/releases/download/java_v11.5/java_tools-v11.5.zip",
+        ],
+    )
+
+    http_archive(
         name = "remote_java_tools_linux",
-        sha256 = "355c27c603e8fc64bb0e2d7f809741f42576d5f4540f9ce28fd55922085af639",
+        sha256 = "ae1eca4546eac6487c6e565f9b409536609c273207220c51e5c94f2a058a5a56",
         urls = [
-            "https://mirror.bazel.build/bazel_java_tools/releases/javac11/v10.5/java_tools_javac11_linux-v10.5.zip",
-            "https://github.com/bazelbuild/java_tools/releases/download/javac11_v10.5/java_tools_javac11_linux-v10.5.zip",
+            "https://mirror.bazel.build/bazel_java_tools/releases/java/v11.5/java_tools_linux-v11.5.zip",
+            "https://github.com/bazelbuild/java_tools/releases/download/java_v11.5/java_tools_linux-v11.5.zip",
         ],
     )
-    maybe(
-        http_archive,
+
+    http_archive(
         name = "remote_java_tools_windows",
-        sha256 = "0b4469ca1a9b3f26cb82fb0f4fd00096f0d839ec2fae097e7bdbb982e3a95a59",
+        sha256 = "36766802f7ec684cecb1a14c122428de6be9784e88419e2ab5912ad4b59a8c7d",
         urls = [
-            "https://mirror.bazel.build/bazel_java_tools/releases/javac11/v10.5/java_tools_javac11_windows-v10.5.zip",
-            "https://github.com/bazelbuild/java_tools/releases/download/javac11_v10.5/java_tools_javac11_windows-v10.5.zip",
+            "https://mirror.bazel.build/bazel_java_tools/releases/java/v11.5/java_tools_windows-v11.5.zip",
+            "https://github.com/bazelbuild/java_tools/releases/download/java_v11.5/java_tools_windows-v11.5.zip",
         ],
     )
-    maybe(
-        http_archive,
+
+    http_archive(
         name = "remote_java_tools_darwin",
-        sha256 = "95aae0a32a170c72a68abb0b9dd6bac7ea3e08c504a5d8c6e8bf7ac51628c98f",
+        sha256 = "792bc1352e736073b152528175ed424687f86a9f6f5f461f07d8b26806762738",
         urls = [
-            "https://mirror.bazel.build/bazel_java_tools/releases/javac11/v10.5/java_tools_javac11_darwin-v10.5.zip",
-            "https://github.com/bazelbuild/java_tools/releases/download/javac11_v10.5/java_tools_javac11_darwin-v10.5.zip",
+            "https://mirror.bazel.build/bazel_java_tools/releases/java/v11.5/java_tools_darwin-v11.5.zip",
+            "https://github.com/bazelbuild/java_tools/releases/download/java_v11.5/java_tools_darwin-v11.5.zip",
         ],
     )
 
-def remote_jdk8_repos():
```

Comment:
Perhaps keep JDK8 remote repos. This is the only location where they are still defined. 

And I was advertising on rules_appengine (which needs JDK8), this is the way to use it without installing JDK8 locally.

But we could also say we don't support JDK8 anymore.

Also, I was advertising that calling for example `remote_jdk8_repos()` from the WORKSPACE will have the JDK8 toolchain ready - defined and registered. Perhaps the code needs to be reorganised a bit to keep this true in the no-bzlmod world.

The user base that's using rules_java and calls in repositories.bzl, is probably really small, though. So I'm not sure, how much is worth spending time on this reorganisation + JDK8.

Reply:
OK, I added remote_jdk8_repos back, also added toolchain registration in it, but I'm not invoking it by default in rules_java_dependencies. PTAL~.